lib/src/message.dart

```diff
@@ -149,55 +149,94 @@ abstract class Message {
   /// changes.
   Message change(
       {Map<String, String> headers, Map<String, Object> context, body});
-}
 
-/// Adds information about encoding to [headers].
-///
-/// Returns a new map without modifying [headers].
-Map<String, String> _adjustHeaders(Map<String, String> headers, Body body) {
-  var sameEncoding = _sameEncoding(headers, body);
-  if (sameEncoding) {
-    if (body.contentLength == null ||
-        getHeader(headers, 'content-length') == body.contentLength.toString()) {
-      return headers ?? const HttpUnmodifiableMap.empty();
-    } else if (body.contentLength == 0 &&
-        (headers == null || headers.isEmpty)) {
-      return const HttpUnmodifiableMap.empty();
+  /// The default set of headers for a message created with no body and no
+  /// explicit headers.
+  static final _defaultHeaders = new HttpUnmodifiableMap<String>(
+      {'content-length': '0'},
+      ignoreKeyCase: true);
```

Comment:
This mirrors what is being done in `shelf`.

https://github.com/dart-lang/shelf/blob/master/lib/src/message.dart#L17-L21
```
final _defaultHeaders = new ShelfUnmodifiableMap<String>(
    {"content-length": "0"},
    ignoreKeyCase: true);
```